`General`: Remove Cypress setup and tests

muradium opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check all tasks by putting an x in the [ ] (don't: [x ], [ x], do: [x]). Remove not applicable tasks and do not leave them unchecked -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
#### General
<!-- Remove tasks that are not applicable for your PR. Please only put the PR into ready for review, if all relevant tasks are checked! -->
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable and you may remove the first checkmark. If you are unsure, please test on the test servers. -->
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context
Cypress E2E testing framework is no longer used in Artemis and was replaced by Playwright

### Description
This PR removes Cypress framework and everything related to it. Following steps were taken:
- Migrate/rename Cypress docker environment files into Playwright
- Move certs from Cypress to Playwright
- Remove Cypress tests
- Remove Cypress setup

### Steps for Testing
- **Code Review**: Ensure that changes either remove everything about Cypress or migrate it to Playwright without breaking other parts of code.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress
<!-- Each PR should be reviewed by at least two other developers. The code, the functionality (= manual test) and the exam mode need to be reviewed. -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->
<!-- All PRs that might affect the exam mode (e.g. change a client component that is also used in the exam mode) need an additional verification that the exam mode still works. -->

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


- **New Features**
  - Transitioned from Cypress to Playwright for E2E testing, including configuration updates and migration of test files.

- **Chores**
  - Updated `.gitignore` to remove Cypress-specific rules.
  - Reconfigured ESLint project paths for improved clarity.
  - Updated Docker setup and environment configuration for Playwright.
  - Simplified script logic for test framework execution.
  - Adjusted IntelliJ module configuration for Artemis server.

- **Documentation**
  - Updated README in test certificates directory to reflect changes from Cypress to Playwright.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

SimonEntholzer commented on docker/test-server-mysql-localci.yml:
```diff
@@ -46,10 +46,10 @@ services:
         volumes:
             - type: bind
-              source: ${NGINX_PROXY_SSL_CERTIFICATE_PATH:-../src/test/cypress/certs/artemis-nginx+4.pem}
+              source: ${NGINX_PROXY_SSL_CERTIFICATE_PATH:-../src/test/playwright/certs/artemis-nginx+4.pem}
```
how do the tests already run on playwright when the yml configuration files seem to still contain cypress?

muradium replied:
Test runs in CI are based on the build plans `docker/playwright-E2E-tests-*`. SSL certificates are currently not used in Playwright (no option to enable https authentication), but updated their paths nevertheless

SimonEntholzer reviewed (commented):
Code changes look good, looking forward to not having to deal with the same files in cypress and playwright any more. Just got one small question

JohannesStoehr reviewed (changes requested):
Code changes look good but can you investigate why there are no e2e tests on Bamboo for this branch? It would be good to see if the tests still run successfully 😅 
Since this branch is already quite old, it might be the case that the tests have to be reenabled by a sys admin.

pzdr7 commented on .idea/runConfigurations/Artemis__Server__LocalVC___LocalCI_.xml:
```diff
@@ -2,12 +2,12 @@
     <option name="ALTERNATIVE_JRE_PATH" value="17" />
-    <module name="Artemis.main" />
+    <module name="de.tum.in.www1.artemis.Artemis.main" />
     <option name="SHORTEN_COMMAND_LINE" value="MANIFEST" />
     <option name="SPRING_BOOT_MAIN_CLASS" value="de.tum.in.www1.artemis.ArtemisApp" />
     <option name="VM_PARAMETERS" value="-XX:+ShowCodeDetailsInExceptionMessages -Duser.country=US -Duser.language=en -Xmx1024m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en --add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED --add-exports jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED --add-exports jdk.compiler/com.sun.tools.javac.parser= ...
     <method v="2">
       <option name="Gradle.BeforeRunTask" enabled="false" tasks="build" externalProjectPath="$PROJECT_DIR$" vmOptions="" scriptParameters="-x webapp -x test -x jacocoTestCoverageVerification -x spotlessCheck -x checkstyleMain -x checkstyleTest" />
     </method>
   </configuration>
-</component>
+</component>
```
Did you mean to change this run config? I ask because none of the other run configs have these changes

muradium replied:
Good catch, that was accidental 😄 Reverted it

krusche commented:
@muradium what is the status of this PR? I kindly ask you to finish it asap so we can merge it. Thank you!

muradium commented:
@krusche I need to solve issues I'm facing with merging develop and implement a small change from feedback. Should be ready to merge today.

pzdr7 reviewed (changes requested):
Please remove the remaining references to cypress from the eslint config - they are causing the client style to fail `ENOENT: no such file or directory, open '/home/runner/work/Artemis/Artemis/src/test/cypress/tsconfig.json'`

----

Some other remarks

- Will the documentation be adjusted in a follow-up? I still see some files regarding Cypress there (https://docs.artemis.cit.tum.de/dev/cypress.html)
- `supporting_scripts/course-setup-quickstart/create_users.py` still refers to cypress users. Do we even need that anymore, given that the playwright tests create the users on their own? 🤔

muradium commented:
@pzdr7 Yes, existing Cypress documentation will be adjusted for Playwright and removed in a separate PR.

Thank for pointing it out, I removed `create_users.py` as well as I don't think it's useful anywhere. For Cypress we also had a separate "Setup users" test, not sure what that script was used for.
